src/custom-surf/api/index.ts

```diff
@@ -287,8 +287,19 @@ export class CustomApiClient extends CustomApiClientInterface {
         return this.fetchJson<ServiceTicket[]>(prefix_cim_dev_uri("surf/cim/tickets"));
     };
 
+    _cimTicketById = (ticket_id: string, show_error_dialog: boolean): Promise<ServiceTicketWithDetails> => {
+        return this.fetchJson<ServiceTicketWithDetails>(
```

Comment:
I know that you didn't add this line; it would be better to have this in the customApiClient. Which also contains the other API calls.

Reply:
> I know that you didn't add this line; it would be better to have this in the customApiClient. Which also contains the other API calls.


This is the CustomApiClient class? Or I'm missing something :)

Reply:
> And maybe even use react query -> than it will be faster

I'll give it a shot

Reply:
I think we should also implement an ETag on the ticket detail endpoint because there's quite a bit of useless traffic 

![image](https://user-images.githubusercontent.com/1482100/216281989-f6a93f95-3543-4051-a90d-11bde7080d20.png)

Not sure yet if I'll make it part of this change

Reply:
Implemented useQuery; for ETag added a note to cim#50